### PR TITLE
CosmosDB enhancements

### LINKED ISSUE #666: azure_rm_cosmosdbaccount module should support setting the MongoDB API server version and other capabilities
##### SUMMARY
I am using the `azure_rm_cosmosdbaccount` module to provision an Azure CosmosDB instance with a MongoDB API.  Currently the module documentation and the code show no way to control the server version of the MongoDB API.  By default, version `3.6` is selected.  I would like to select the latest server version, which is `4.0`.  Additionally, I would like to be able to configure additional features, such as the `Serverless` capacity mode.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
`azure_rm_cosmosdbaccount`

##### ADDITIONAL INFORMATION
The module parameters for `azure_rm_cosmosdbaccount` seem to mirror the Azure CosmosDB properties, so my idea would be to add parameters mirroring the `apiProperties` and `capabilities` properties.  Here are property examples from an Azure CosmosDB JSON response:

```json
"apiProperties": {
    "serverVersion": "4.0"
}
...
"capabilities": [
    {
        "name": "EnableMongo"
    },
    {
        "name": "EnableServerless"
    }
 ]
```

Based on this, here is an example of the module parameters that could be added:
```yaml
api_properties
  server_version: 4.0
...
capabilities:
  - name: EnableMongo
  - name: EnableServerless
```

### PR DESCRIPTION
**Pull request by l3ender**
##### SUMMARY
This PR adds additional support for CosmosDB features:
* Ability to specify database version for MongoDB (`mongo_version`)--added in [SDK version 0.15.0](https://github.com/Azure/azure-sdk-for-python/releases/tag/azure-mgmt-cosmosdb_0.15.0).
* Ability to enable free tier model (`enable_free_tier`)--added in [SDK version 0.15.0](https://github.com/Azure/azure-sdk-for-python/releases/tag/azure-mgmt-cosmosdb_0.15.0). Note: as of 2021, there can only be one free tier mongo instance per Azure subscription.
* Ability to enable/disable public network access to the server (`public_network_access`)--added in [SDK version 0.13.0](https://github.com/Azure/azure-sdk-for-python/releases/tag/azure-mgmt-cosmosdb_0.13.0).
* Due to SDK dependency updates, the `ip_range_filter` parameter is being deprecated in favor of the new parameter `ip_rules`.

This PR partially resolves https://github.com/ansible-collections/azure/issues/666.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- azure_rm_cosmosdbaccount
- azure_rm_cosmosdbaccount_info

##### ADDITIONAL INFORMATION

Using the following playbook at the root of this repo:

```yaml
---
- name: "Playbook for testing."
  hosts: "localhost"
  connection: "local"
  gather_facts: false
  vars:
    resource_group: "automated-testing"
    resource_group_secondary: "automated-testing-secondary"
  collections:
    - azure.azcollection

  tasks:
    - name: "Include tests"
      include_tasks: "tests/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml"
```

<details>
<summary>Click to view playbook output</summary>

```bash
-> ansible-playbook sandbox.yml -v
No config file found; using defaults

PLAY [Playbook for testing.] ***************************************************

TASK [Include tests] ***********************************************************
included: /Users/ross/repos/azure/tests/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml for localhost

TASK [Prepare random number] ***************************************************
ok: [localhost] => {"ansible_facts": {"dbname": "cosmosef45326637", "subnetname": "subnetef45326192", "vnname": "vnef45326406"}, "changed": false}

TASK [Create virtual network] **************************************************
changed: [localhost] => {"changed": true, "check_mode": false, "state": {"address_prefixes": ["10.1.0.0/16", "172.100.0.0/16"], "dns_servers": ["127.0.0.1", "127.0.0.3"], "etag": "W/\"4108a652-67a1-48d8-8562-534d5cd1cc4a\"", "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406", "location": "southcentralus", "name": "vnef45326406", "provisioning_state": "Succeeded", "tags": null, "type": "Microsoft.Network/virtualNetworks"}}

TASK [Add subnet] **************************************************************
changed: [localhost] => {"changed": true, "state": {"address_prefix": "10.1.0.0/24", "address_prefixes": null, "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406/subnets/subnetef45326192", "name": "subnetef45326192", "network_security_group": {}, "private_endpoint_network_policies": "Enabled", "private_link_service_network_policies": "Enabled", "provisioning_state": "Succeeded", "route_table": {}}}

TASK [Create instance of Database Account -- check mode] ***********************
changed: [localhost] => {"changed": true}

TASK [Assert the resource instance is well created] ****************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create instance of Database Account] *************************************
changed: [localhost] => {"changed": true, "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637"}

TASK [Assert the resource instance is well created] ****************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create again instance of Database Account] *******************************
ok: [localhost] => {"changed": false, "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637"}

TASK [Assert the state has not changed] ****************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create again instance of Database Account -- change something] ***********
changed: [localhost] => {"changed": true, "compare": "changed [/enable_automatic_failover] True != False", "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637"}

TASK [Assert the state has changed] ********************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Get facts of single account] *********************************************
ok: [localhost] => {"accounts": [{"consistency_policy": {"default_consistency_level": "session", "max_interval_in_seconds": 5, "max_staleness_prefix": 100}, "database_account_offer_type": "Standard", "document_endpoint": "https://cosmosef45326637.documents.azure.com:443/", "enable_automatic_failover": true, "enable_cassandra": false, "enable_free_tier": false, "enable_gremlin": false, "enable_multiple_write_locations": false, "enable_table": false, "failover_policies": [{"failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia"}, {"failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus"}], "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637", "ip_range_filter": "", "ip_rules": [], "is_virtual_network_filter_enabled": true, "kind": "global_document_db", "location": "eastasia", "mongo_version": null, "name": "cosmosef45326637", "provisioning_state": "Succeeded", "public_network_access": "Enabled", "read_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}, {"document_endpoint": "https://cosmosef45326637-westus.documents.azure.com:443/", "failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus", "provisioning_state": "Succeeded"}], "resource_group": "automated-testing", "tags": {}, "virtual_network_rules": [{"id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406/subnets/subnetef45326192", "ignore_missing_vnet_service_endpoint": true}], "write_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}]}], "changed": false}

TASK [Assert that facts are returned] ******************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Get facts with keys] *****************************************************
ok: [localhost] => {"accounts": [{"consistency_policy": {"default_consistency_level": "session", "max_interval_in_seconds": 5, "max_staleness_prefix": 100}, "database_account_offer_type": "Standard", "document_endpoint": "https://cosmosef45326637.documents.azure.com:443/", "enable_automatic_failover": true, "enable_cassandra": false, "enable_free_tier": false, "enable_gremlin": false, "enable_multiple_write_locations": false, "enable_table": false, "failover_policies": [{"failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia"}, {"failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus"}], "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637", "ip_range_filter": "", "ip_rules": [], "is_virtual_network_filter_enabled": true, "kind": "global_document_db", "location": "eastasia", "mongo_version": null, "name": "cosmosef45326637", "primary_master_key": "3sRzCEsGDfcFfFFfM5pRX1eGF8zPzg8unILjUYaOtcWXOW9qYg9iC84UC1HQoVI0i3bhYt6SdUa36VjBchBytA==", "primary_readonly_master_key": "0qm0LRpHOdmdm11kJpc5UYKG4rdDX0X1jdWTqTd1wWElnQogzNr6TjeFjue7GwU6fYYkhnhzOC9Z4eDQrhnnPw==", "provisioning_state": "Succeeded", "public_network_access": "Enabled", "read_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}, {"document_endpoint": "https://cosmosef45326637-westus.documents.azure.com:443/", "failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus", "provisioning_state": "Succeeded"}], "resource_group": "automated-testing", "secondary_master_key": "lYKzMO1uu8hEP1UWDl3PSXjvoeh1fS6QYcADlO9qlevgTuupp0XHCaWyI5vrrS3qGmmHfykN36HgJbhsCAv2Ag==", "secondary_readonly_master_key": "Jd8aTrlOd4o35B90busFYwHiOREhQDmhBfZ980w9qPEgxqyLnH7tEaE26uK49NxigLK8MoN9p9FdkQohViUVBA==", "tags": {}, "virtual_network_rules": [{"id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406/subnets/subnetef45326192", "ignore_missing_vnet_service_endpoint": true}], "write_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}]}], "changed": false}

TASK [Assert that facts are returned] ******************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Get facts with readonly keys] ********************************************
ok: [localhost] => {"accounts": [{"connection_strings": {"connection_strings": [{"connection_string": "AccountEndpoint=https://cosmosef45326637.documents.azure.com:443/;AccountKey=3sRzCEsGDfcFfFFfM5pRX1eGF8zPzg8unILjUYaOtcWXOW9qYg9iC84UC1HQoVI0i3bhYt6SdUa36VjBchBytA==;", "description": "Primary SQL Connection String"}, {"connection_string": "AccountEndpoint=https://cosmosef45326637.documents.azure.com:443/;AccountKey=lYKzMO1uu8hEP1UWDl3PSXjvoeh1fS6QYcADlO9qlevgTuupp0XHCaWyI5vrrS3qGmmHfykN36HgJbhsCAv2Ag==;", "description": "Secondary SQL Connection String"}, {"connection_string": "AccountEndpoint=https://cosmosef45326637.documents.azure.com:443/;AccountKey=0qm0LRpHOdmdm11kJpc5UYKG4rdDX0X1jdWTqTd1wWElnQogzNr6TjeFjue7GwU6fYYkhnhzOC9Z4eDQrhnnPw==;", "description": "Primary Read-Only SQL Connection String"}, {"connection_string": "AccountEndpoint=https://cosmosef45326637.documents.azure.com:443/;AccountKey=Jd8aTrlOd4o35B90busFYwHiOREhQDmhBfZ980w9qPEgxqyLnH7tEaE26uK49NxigLK8MoN9p9FdkQohViUVBA==;", "description": "Secondary Read-Only SQL Connection String"}]}, "consistency_policy": {"default_consistency_level": "session", "max_interval_in_seconds": 5, "max_staleness_prefix": 100}, "database_account_offer_type": "Standard", "document_endpoint": "https://cosmosef45326637.documents.azure.com:443/", "enable_automatic_failover": true, "enable_cassandra": false, "enable_free_tier": false, "enable_gremlin": false, "enable_multiple_write_locations": false, "enable_table": false, "failover_policies": [{"failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia"}, {"failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus"}], "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637", "ip_range_filter": "", "ip_rules": [], "is_virtual_network_filter_enabled": true, "kind": "global_document_db", "location": "eastasia", "mongo_version": null, "name": "cosmosef45326637", "primary_readonly_master_key": "0qm0LRpHOdmdm11kJpc5UYKG4rdDX0X1jdWTqTd1wWElnQogzNr6TjeFjue7GwU6fYYkhnhzOC9Z4eDQrhnnPw==", "provisioning_state": "Succeeded", "public_network_access": "Enabled", "read_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}, {"document_endpoint": "https://cosmosef45326637-westus.documents.azure.com:443/", "failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus", "provisioning_state": "Succeeded"}], "resource_group": "automated-testing", "secondary_readonly_master_key": "Jd8aTrlOd4o35B90busFYwHiOREhQDmhBfZ980w9qPEgxqyLnH7tEaE26uK49NxigLK8MoN9p9FdkQohViUVBA==", "tags": {}, "virtual_network_rules": [{"id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406/subnets/subnetef45326192", "ignore_missing_vnet_service_endpoint": true}], "write_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}]}], "changed": false}

TASK [Assert that facts are returned] ******************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [List acounts by resource group] ******************************************
ok: [localhost] => {"accounts": [{"consistency_policy": {"default_consistency_level": "session", "max_interval_in_seconds": 5, "max_staleness_prefix": 100}, "database_account_offer_type": "Standard", "document_endpoint": "https://cosmosef45326637.documents.azure.com:443/", "enable_automatic_failover": true, "enable_cassandra": false, "enable_free_tier": false, "enable_gremlin": false, "enable_multiple_write_locations": false, "enable_table": false, "failover_policies": [{"failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia"}, {"failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus"}], "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637", "ip_range_filter": "", "ip_rules": [], "is_virtual_network_filter_enabled": true, "kind": "global_document_db", "location": "eastasia", "mongo_version": null, "name": "cosmosef45326637", "provisioning_state": "Succeeded", "public_network_access": "Enabled", "read_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}, {"document_endpoint": "https://cosmosef45326637-westus.documents.azure.com:443/", "failover_priority": 1, "id": "cosmosef45326637-westus", "name": "westus", "provisioning_state": "Succeeded"}], "resource_group": "automated-testing", "tags": {}, "virtual_network_rules": [{"id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406/subnets/subnetef45326192", "ignore_missing_vnet_service_endpoint": true}], "write_locations": [{"document_endpoint": "https://cosmosef45326637-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}]}], "changed": false}

TASK [Assert that facts are returned] ******************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Delete instance of Database Account -- check mode] ***********************
changed: [localhost] => {"changed": true}

TASK [Assert the state has changed] ********************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Delete instance of Database Account] *************************************
changed: [localhost] => {"changed": true}

TASK [Assert the state has changed] ********************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Delete unexisting instance of Database Account] **************************
ok: [localhost] => {"changed": false}

TASK [Assert the state has not changed] ****************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create 4.0 Mongo free-tier instance with public networking disabled] *****
changed: [localhost] => {"changed": true, "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637-free4"}

TASK [Assert the state has not changed] ****************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Get facts for free tier account] *****************************************
ok: [localhost] => {"accounts": [{"consistency_policy": {"default_consistency_level": "session", "max_interval_in_seconds": 5, "max_staleness_prefix": 100}, "database_account_offer_type": "Standard", "document_endpoint": "https://cosmosef45326637-free4.documents.azure.com:443/", "enable_automatic_failover": true, "enable_cassandra": false, "enable_free_tier": true, "enable_gremlin": false, "enable_multiple_write_locations": false, "enable_table": false, "failover_policies": [{"failover_priority": 0, "id": "cosmosef45326637-free4-eastasia", "name": "eastasia"}, {"failover_priority": 1, "id": "cosmosef45326637-free4-westus", "name": "westus"}], "id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.DocumentDB/databaseAccounts/cosmosef45326637-free4", "ip_range_filter": "1.1.1.1,2.2.2.2/28", "ip_rules": ["1.1.1.1", "2.2.2.2/28"], "is_virtual_network_filter_enabled": true, "kind": "mongo_db", "location": "eastasia", "mongo_version": "4.0", "name": "cosmosef45326637-free4", "provisioning_state": "Succeeded", "public_network_access": "Disabled", "read_locations": [{"document_endpoint": "https://cosmosef45326637-free4-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-free4-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}, {"document_endpoint": "https://cosmosef45326637-free4-westus.documents.azure.com:443/", "failover_priority": 1, "id": "cosmosef45326637-free4-westus", "name": "westus", "provisioning_state": "Succeeded"}], "resource_group": "automated-testing", "tags": {}, "virtual_network_rules": [{"id": "/subscriptions/15d7b290-260a-4ea7-a138-37b3f781f644/resourceGroups/automated-testing/providers/Microsoft.Network/virtualNetworks/vnef45326406/subnets/subnetef45326192", "ignore_missing_vnet_service_endpoint": true}], "write_locations": [{"document_endpoint": "https://cosmosef45326637-free4-eastasia.documents.azure.com:443/", "failover_priority": 0, "id": "cosmosef45326637-free4-eastasia", "name": "eastasia", "provisioning_state": "Succeeded"}]}], "changed": false}

TASK [Assert that facts are returned] ******************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP *********************************************************************
localhost                  : ok=30   changed=8    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```
</details>

### CODE REVIEW

**Comment by l3ender** [comment]
@Fred-sun Can you please take a look? Thank you!

**Comment by Fred-sun** [comment]
@l3ender Thank you for your contribution. We will complete the review and push forward the merger as soon as possible.

**Comment by Fred-sun** [comment]
@l3ender Can you add test cases for the newly added parameters? Thank you very much!

**Comment by l3ender** [comment]
@Fred-sun I added test cases and updated the description on this PR. Please take a look--thank you!

**Comment by Fred-sun** [comment]
@l3ender Can you share the results returned by your test case? I add test cases, but always get a create error when creating them. Thank you very much!

**Comment by l3ender** [comment]
@Fred-sun I have verbose output of test case run in PR description.

Note there can only be one free tier mongo instance per Azure subscription. If one already exists, you will get an error like following when trying to create:
> Error creating the Database Account instance: 400 Client Error: BadRequest for url: https://management.azure.com/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.DocumentDB/databaseAccounts/xxx?api-version=2020-04-01

**Comment by l3ender** [comment]
Hi @Fred-sun, is there any update on this PR?

My assumption is the PR was put on hold because of the SDK update, but I haven't seen any movement for other PRs related to SDK updates. I think a change like this (updating only a single SDK library) is a safe approach which also contributes to the overall goal of updating all SDK versions (a large effort to do in single PR).

Thanks!

**Comment by Fred-sun** [comment]
@l3ender I'm working in it! Thank you very much!

**Comment by l3ender** [comment]
Hi @Fred-sun, anything else needed for this PR? Thank you!

File: plugins/modules/azure_rm_cosmosdbaccount.py [hunk 1]
```diff
@@ -79,11 +79,23 @@
...
         description:
-            - Cosmos DB Firewall support. This value specifies the set of IP addresses or IP address ranges.
+            - (deprecated) Cosmos DB Firewall support. This value specifies the set of IP addresses or IP address ranges.
             - In CIDR form to be included as the allowed list of client IPs for a given database account.
             - IP addresses/ranges must be comma separated and must not contain any spaces.
+            - This value has been deprecated, and will be removed in a later version. Use I(ip_rules) instead.
+    ip_rules:
+        description:
+            - The IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs.
+        type: list
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_cosmosdbaccount.py:97]:
```suggestion
        type: list
        elements: str
```

File: tests/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml [hunk 2]
```diff
@@ -247,6 +253,54 @@
...
       - db2name in (output.accounts | map(attribute='name'))
 
+- name: Create 4.0 Mongo free-tier instance with public networking disabled
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}-free4"
+    location: eastasia
+    kind: mongo_db
+    mongo_version: "4.0"
+    enable_free_tier: true
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml]: Free tier is not supported for Internal subscriptions
```suggestion
    enable_free_tier: false
```

File: tests/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml [hunk 3]
```diff
@@ -247,6 +253,54 @@
...
+- name: Get facts for free tier account
+  azure_rm_cosmosdbaccount_info:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}-free4"
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.accounts[0]['mongo_version'] == '4.0'
+      - output.accounts[0]['enable_free_tier'] == true
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml]:
```suggestion
      - output.accounts[0]['enable_free_tier'] == false
```

**Comment by Fred-sun** [comment]
@l3ender Please make some changes to the test case. Thank you very much!

**Comment by l3ender** [comment]
@Fred-sun, I have updated accordingly. Thank you!

**Comment by xuzhang3** [comment]
Thanks @l3ender LGTM

**Pull request merged**